src/Symfony/Component/Notifier/Bridge/Expo/ExpoTransport.php

```diff
@@ -0,0 +1,108 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Expo;
+
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Imad ZAIRIG <https://github.com/zairigimad>
+ *
+ * @experimental in 5.4
+ */
+final class ExpoTransport extends AbstractTransport
+{
+    protected const HOST = 'exp.host/--/api/v2/push/send';
+
+    /** @var string */
+    private $token;
+
+    public function __construct(string $token = null, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->token = $token;
+        $this->client = $client;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('expo://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        $endpoint = sprintf('https://%s', $this->getEndpoint());
+        $options = ($opts = $message->getOptions()) ? $opts->toArray() : [];
+        if (!isset($options['to'])) {
+            $options['to'] = $message->getRecipientId();
+        }
+        if (null === $options['to']) {
+            throw new InvalidArgumentException(sprintf('The "%s" transport required the "to" option to be set.', __CLASS__));
+        }
+        $options['body'] = $message->getSubject();
+        $options['data'] = $options['data'] ?? [];
+
+        $response = $this->client->request('POST', $endpoint, [
+            'headers' => [
+                'Authorization' => $this->token ? sprintf('Bearer %s', $this->token) : null,
+            ],
+            'json' => array_filter($options),
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote Expo server.', $response, 0, $e);
+        }
+
+        $contentType = $response->getHeaders(false)['content-type'][0] ?? '';
+        $jsonContents = 0 === strpos($contentType, 'application/json') ? $response->toArray(false) : null;
+
```

Comment:
do we really need to check the content-type?

What's about wrapping `$response->toArray(false)` in a try//catch block?

Reply:
Actually, i tried to follow the same implementation as firebase notifier here